packages/shorebird_cli/lib/src/platform/apple.dart

```diff
@@ -26,7 +27,30 @@ enum ApplePlatform {
 
 /// A record containing the exit code and optionally link percentage
 /// returned by `runLinker`.
-typedef LinkResult = ({int exitCode, double? linkPercentage});
+@immutable
+class LinkResult {
+  /// Creates a new [LinkResult] representing failure.
+  const LinkResult.failure()
```

Comment:
wouldn't we want to have an error reason (e.g. from `stderr`) in the case of a failure?

Reply:
We certainly could.  I was just matching what the existing code did, just moving it from being a record type to an explicit class with a named constructor (with appropriate defaults).